### PR TITLE
[Feature] support mim

### PR DESCRIPTION
**Pull request by xiexinch**
- add MIM introduction in README
- modify setup.py to support mim as mmseg entry point

### CODE REVIEW

**Comment by xiexinch** [comment]
@zhouzaida

**Review by Junjun2016 — APPROVED** [review]

File: MANIFEST.in [hunk 1]
```diff
@@ -1,5 +1,5 @@
-
 include requirements/*.txt
-include mmseg/model-index.yml
-recursive-include mmseg/configs *.py *.yml
-recursive-include mmseg/tools *.sh *.py
+include mmseg/.mim/model-index.yml
+recursive-include mmseg/.mim/configs *.py *.yml
+recursive-include mmseg/.mim/tools *.py *.sh
+recursive-include mmseg/.mim/demo *.py
```
Comment (zhouzaida) [inline MANIFEST.in:5]: This line should be removed

File: setup.py [hunk 2]
```diff
@@ -92,7 +97,56 @@ def gen_packages_items():
...
+        # installed by `pip install -e .`
+        mode = 'symlink'
+    elif 'sdist' in sys.argv or 'bdist_wheel' in sys.argv:
+        # installed by `pip install .`
+        # or create source distribution by `python setup.py sdist`
+        mode = 'copy'
+    else:
+        return
+
+    filenames = ['tools', 'configs', 'demo', 'model-index.yml']
```
Comment (zhouzaida) [inline setup.py:119]:
```suggestion
    filenames = ['tools', 'configs', 'model-index.yml']
```

**Review by zhouzaida — APPROVED** [review]
LGTM

**Pull request merged**